mistralrs-core/src/pipeline/inputs_processor.rs

```diff
@@ -148,6 +262,47 @@ pub mod text_models_inputs_processor {
             position_ids.push(seq.len());
 
             seqs_tensors.push(Tensor::new(ctxt, device).unwrap().unsqueeze(0).unwrap());
+
+            if let Some(ref paged_attn_metadata) = paged_attn_metadata {
+                let table = paged_attn_metadata.block_tables.get(seq.id()).unwrap();
+
+                let table = table
+                    .iter()
+                    .map(|block| block.deref_mut().block_id)
+                    .collect::<Vec<_>>();
+
+                let block_number = if start_pos / paged_attn_metadata.block_size >= table.len() {
+                    table.last().unwrap() //position exceed! use last position; TODO (bug fix)
```

Comment:
@guoqingbao I was wondering what you meant with this comment?

Reply:
I encounter the problem of start_pos / paged_attn_metadata.block_size larger than the size of block table in candle-vllm. It seems that the calculation of block table size has some bugs. That's why I padded the prompt tokens into odd number.

Reply:
We don't do any padding of the prompt to an odd number, do you think this is a problem here? Perhaps a solution is to make sure we allocate one extra physical token block?

Reply:
From my experience in candle-vllm, the program will be crashed in some cases if no padding to odd number. I think the root cause lies in the allocation of block tables. Perhaps extra block table can address the problem. I haven't figure out how to solve it permanently in an elegant way.

Reply:
Yeah, this is tricky. I took a look at the code which manages the logical and physical token blocks (the BlockEngine :) ), it appears to do what we want: add the token to a non-full block or append a new one and start filling that up.

Specifically, let's say block_size = 32, and we add 32 prompt tokens: we will have 1 logical token block, so when `BlockEngine::allocate` is called, we make one physical token block. So `tables.len == 1`. But `start_pos == 32-1 == 31`, so `31 / 32 == 0 < 1`. 

This looks like an edge case, I can't think of any case where this would be true. Perhaps I am not understanding the error case correctly? Would it be correct to assume that this occurs any time `start_pos` is even if there is an even number of prompt tokens (this check is not in the decoding preparation, so maybe that is the case)?

Reply:
This problem occurs especially when the start_pos is divisible by the block size (e.g., 32). Normally, when the start_pos is divisible by the block size (say 64 / 32), the number of block tables should be at least 3, but that is not the case. So that I padded it to the odd number to ensure extra block table in the corner case. Perhaps we are not taking account the start token when calculating the number of block tables, for example, a request of 100 tokens would occupy 101 token space in the block tables.

Reply:
> This problem occurs especially when the start_pos is divisible by the block size (e.g., 32). Normally, when the start_pos is divisible by the block size (say 64 / 32), the number of block tables should be at least 3, but that is not the case. So that I padded it to the odd number to ensure extra block table in the corner case. Perhaps we are not taking account the start token when calculating the number of block tables, for example, a request of 100 tokens would occupy 101 token space in the block tables before decoding.

Reply:
Ah ok. I reproduced the error :)

Reply:
> Ah ok. I reproduced the error :)

Have you checked the allocation of block tables on the basis of the number of input tokens (prompt)? I think the problem is there, you may pad the input tokens to the size of divisible by the block size, it will be easier to reproduce and debug. :)